src/TwigComponent/src/ComponentRenderer.php

```diff
@@ -38,14 +39,21 @@ public function __construct(
     ) {
     }
 
-    public function createAndRender(string $name, array $props = []): string
+    /**
+     * Allow the render process to be short-circuited.
+     */
```

Comment:
Dispatch an event before rendering to give an opportunity to short-circuit the rendering process.